internal/provider/folder_resource_test.go

```diff
@@ -0,0 +1,209 @@
+package provider
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"net/http"
+	"os"
+	"testing"
+
+	"github.com/cockroachdb/cockroach-cloud-sdk-go/pkg/client"
+	mock_client "github.com/cockroachdb/terraform-provider-cockroach/mock"
+	"github.com/golang/mock/gomock"
+	"github.com/google/uuid"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+)
+
+// TestAccFolderResource attempts to create, check, update, and destroy real
+// folders. It will be skipped if TF_ACC isn't set.
+func TestAccFolderResource(t *testing.T) {
```

Comment:
For this test, we'll need to enable folders for the test org.

Reply:
Done.

Reply:
Thanks! I ran these tests using my service account in the test org, then updated the acceptance test service account's roles to include the org-level FOLDER_ADMIN role (and other FGAC admin roles). 

<img width="713" alt="Screenshot 2023-08-03 at 2 37 11 PM" src="https://github.com/cockroachdb/terraform-provider-cockroach/assets/5688006/72656629-6e33-4741-8c05-1dbb1330d4f7">

The serverless cluster test is also failing for the main branch, so I think it's unrelated to changes in this PR. 
<img width="950" alt="Screenshot 2023-08-03 at 3 52 47 PM" src="https://github.com/cockroachdb/terraform-provider-cockroach/assets/5688006/3b2b7ea8-72b5-4ae5-a1f5-2078610392f3">

Reply:
Now the tests all pass, thanks for fixing that one!